Formula/go-critic.rb

```diff
@@ -0,0 +1,34 @@
+class GoCritic < Formula
+  desc "Opinionated Go source code linter"
+  homepage "https://go-critic.com"
+  url "https://github.com/go-critic/go-critic/archive/refs/tags/v0.6.1.tar.gz"
+  sha256 "5d20e626a2803317abb80a8705f80b3ee107f75fd097ca8efaff73173409dd52"
+  license "MIT"
+  head "https://github.com/go-critic/go-critic.git", branch: "master"
+
+  depends_on "go"
+
+  def install
+    ldflags = "-s -w"
+    ldflags += " -X main.Version=v#{version}" unless build.head?
+    system "go", "build", "-trimpath", "-ldflags", ldflags, "-o", bin/"gocritic", "./cmd/gocritic"
```

Comment:
```suggestion
    system "go", "build", *std_go_args(ldflags: ldflags), "./cmd/gocritic"
```

Reply:
The binary itself should be named `gocritic` (in all their upstream documentation, and in the few package managers that carry it), as opposed to `go-critic`. The only thing from `std_go_args` that we can use without any modification is `-trimpath`. This is why I chose not to use `std_go_args`.